File: app/src/main/java/com/fredhappyface/brainf/ActivitySettings.java
package com.fredhappyface.brainf;

import android.content.SharedPreferences;
import android.os.Build;
import android.os.Bundle;
import android.widget.LinearLayout;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.preference.PreferenceManager;
import android.view.View;
import com.google.android.material.card.MaterialCardView;


public class ActivitySettings extends ActivityAbstract {

    @Override
    protected final void onCreate(@Nullable final Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_settings);

        final LinearLayout themeChoices = findViewById(R.id.theme);
        sharedPreferences = PreferenceManager.getDefaultSharedPreferences(this);
        currentTheme = sharedPreferences.getInt("theme", 3);

        final MaterialCardView cardView = (MaterialCardView) themeChoices.getChildAt(currentTheme);
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
            cardView.setStrokeWidth((int) (3 * getApplicationContext().getResources().getDisplayMetrics().density));
            cardView.setStrokeColor(getResources().getColor(R.color.red, getTheme()));
        }

    }

    public final void changeTheme(@NonNull final View view) {
        int idx = 3;
        switch (view.getId()) {
            case R.id.radioLight:
                idx = 0;
                break;
            case R.id.radioDark:
                idx = 1;
                break;
            case R.id.radioBlack:
                idx = 2;
                break;
        }
        final SharedPreferences.Editor editor = sharedPreferences.edit();
        editor.putInt("theme", idx);
        editor.apply();
        recreate();
    }

}

File: app/src/main/java/com/fredhappyface/brainf/ActivitySettings.kt
package com.fredhappyface.brainf

import android.content.SharedPreferences.Editor
import android.os.Bundle
import android.view.View
import android.widget.LinearLayout
import androidx.preference.PreferenceManager
import com.google.android.material.card.MaterialCardView

/**
 * ActivitySettings class inherits from the ActivityThemable class - provides the settings view
 */
class ActivitySettings : ActivityThemable() {
	/**
	 * Override the onCreate method from ActivityThemable adding the activity_settings view and selecting
	 * the current theme
	 *
	 * @param savedInstanceState saved state
	 */
	override fun onCreate(savedInstanceState: Bundle?) {
		super.onCreate(savedInstanceState)
		setContentView(R.layout.activity_settings)
		val sharedPreferences = PreferenceManager.getDefaultSharedPreferences(this)
		// Theme apply border to selected
		val cardView = findViewById<LinearLayout>(R.id.theme).getChildAt(
			sharedPreferences.getInt("theme", 3)
		) as MaterialCardView
		cardView.strokeWidth = (3 * applicationContext.resources.displayMetrics.density).toInt()
		cardView.strokeColor = resources.getColor(R.color.red, theme)
	}

	/**
	 * Compare view.id of the radio button selected to set the theme and recreate the activity
	 *
	 * @param view
	 */
	fun changeTheme(view: View) {
		val editor: Editor = this.sharedPreferences.edit()
		editor.putInt(
			"theme", when (view.id) {
				R.id.radioLight -> 0
				R.id.radioDark -> 1
				R.id.radioBlack -> 2
				else -> 3
			}
		)
		editor.apply()
		recreate()
	}
}
